### PR TITLE
Add create renter page & send renter form

### PR DESCRIPTION
**Pull request by dimalian**
# Changes proposed

- Add form to createRenter page
- Send tenant form

## Demo
Screenshot Before:

image

Screenshot After:

image

## New components

- GoogleAutocomplete component

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,2 @@
+REACT_APP_API_URL=http://localhost:8080
+REACT_APP_GOOGLE_PLACE_API=<Set google place api key>
```
Comment (irenlian) [inline .env.example]: We'll need to use environment variables in Heroku instead `.env` file.

File: package.json [hunk 2]
```diff
@@ -24,14 +24,25 @@
...
+    "@material-ui/icons": "^4.9.1",
+    "@material-ui/lab": "^4.0.0-alpha.56",
+    "@material-ui/styles": "^4.10.0",
+    "@types/autosuggest-highlight": "^3.1.0",
+    "@types/phone": "^2.4.0",
+    "autosuggest-highlight": "^3.1.1",
     "axios": "^0.19.2",
     "classnames": "^2.2.6",
     "connected-react-router": "^6.8.0",
+    "dotenv": "^8.2.0",
```
Comment (irenlian) [inline package.json]: I'm not sure we need it

File: src/common/reportError.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export default function reportError(e: string): void {
+  // eslint-disable-next-line no-console
+  console.error(e);
+}
```
Comment (irenlian) [inline src/common/reportError.ts]: I believe you need to use another type of Error instance.

File: src/components/NewRenter/CreateRenter.tsx [hunk 4]
```diff
@@ -1,9 +1,360 @@
...
+
+const autocompleteService = { current: null };
+
+const defaultValidationForm = {
+  phone: '',
+};
+
+interface PlaceType {
+  description: string;
+  place_id: string;
```
Comment (irenlian) [inline src/components/NewRenter/CreateRenter.tsx]: I'd suggest you move this interface outside.

File: src/components/NewRenter/CreateRenter.tsx [hunk 5]
```diff
@@ -1,9 +1,360 @@
...
+            label={intl.formatMessage(messages.userPhone)}
+            required
+            helperText={validationForm.phone}
+          />
+          <TextField name="social" label={intl.formatMessage(messages.social)} />
+        </Box>
+        <Box display="flex" justifyContent="space-between">
+          <Autocomplete
+            id="google-map"
+            autoHighlight
```
Comment (irenlian) [inline src/components/NewRenter/CreateRenter.tsx]: Perhaps, we could make a separate component with autocomplete with its own logic.

File: src/components/NewRenter/CreateRenter.tsx [hunk 6]
```diff
@@ -1,9 +1,360 @@
...
+                matches.map((match: any) => [match.offset, match.offset + match.length]),
+              );
+
+              return (
+                <Grid container alignItems="center">
+                  <Grid item xs>
+                    {parts.map((part, index) => (
+                      <span
+                        // eslint-disable-next-line react/no-array-index-key
+                        key={index}
```
Comment (irenlian) [inline src/components/NewRenter/CreateRenter.tsx]: I'd suggest a unique key generator.

File: src/components/NewRenter/CreateRenter.tsx [hunk 7]
```diff
@@ -1,9 +1,360 @@
...
+                        style={{
+                          fontWeight: part.highlight ? 700 : 400,
+                        }}
```
Comment (irenlian) [inline src/components/NewRenter/CreateRenter.tsx]: We should stick to one styling practice. SCSS was chosen as a major one.

File: src/components/NewRenter/CreateRenter.tsx [hunk 8]
```diff
@@ -1,9 +1,360 @@
...
+          <FormControl className={classes.formControl}>
+            <InputLabel id="select-currency-label"><FormattedMessage {...messages.currency} /></InputLabel>
+            <Select
+              name="currency"
+              labelId="select-currency-label"
+              defaultValue="USD"
+            >
+              <MenuItem value="USD">USD</MenuItem>
+              <MenuItem value="UAH">UAH</MenuItem>
+              <MenuItem value="EUR">EUR</MenuItem>
```
Comment (irenlian) [inline src/components/NewRenter/CreateRenter.tsx]: Let's use enum here

File: src/components/NewRenter/CreateRenter.tsx [hunk 9]
```diff
@@ -1,9 +1,360 @@
...
+
+const CreateRenter: React.FunctionComponent = () => {
+  const intl = useIntl();
+  const classes = useStyles();
+  const [validationForm, setValidationForm] = useState(defaultValidationForm);
+  const [locationValue, setLocationValue] = React.useState<PlaceType | null>(null);
+  const [inputLocationValue, setInputLocationValue] = React.useState('');
+  const [locationOptions, setLocationOptions] = React.useState<PlaceType[]>([]);
+  const [isSendingForm, setIsSendingForm] = React.useState(false);
+  const [openAlert, setOpenAlert] = React.useState<Color>();
```
Comment (irenlian) [inline src/components/NewRenter/CreateRenter.tsx]: I offer to use this like boolean as we have just one type of alert.

File: src/config.ts [hunk 10]
```diff
@@ -0,0 +1,8 @@
+export default {
+  api: {
+    url: process.env.REACT_APP_API_URL,
+  },
+  google: {
+    placeApi: process.env.REACT_APP_GOOGLE_PLACES_API,
+  },
```
Comment (irenlian) [inline src/config.ts]: Do you really need to nest them?

**Review by irenlian — COMMENTED** [review]
A lot of work is done! It'd be cool to have screenshots in PR.

File: package.json (hunk 2 above)
Comment (dimalian) [reply package.json]: We need it. Without dotenv our env file doesn't pull up

File: .env.example (hunk 1 above)
Comment (dimalian) [reply .env.example]: We will set it in future

File: package.json (hunk 2 above)
Comment (irenlian) [reply package.json]: We'll need to keep it only for development env, I guess.

File: src/components/NewRenter/CreateRenter.tsx [hunk 11]
```diff
@@ -1,9 +1,223 @@
...
+const defaultValidationForm = {
+  phone: '',
+};
+
+const CURRENCY = {
+  USD: 'USD',
+  EUR: 'EUR',
+  UAH: 'UAH',
+  RUB: 'RUB',
+};
```
Comment (irenlian) [inline src/components/NewRenter/CreateRenter.tsx]: Can we convert it to enum and move outside?

**Review by irenlian — APPROVED** [review]

**Pull request merged**